### PR TITLE
Build pipeline fix, drop Drupal <10.1 support, flag PHP 8.2 support

### LINKED ISSUE #93: PHP 8.2 compatibility
FTR, PHP 8.2 compatibility can be only marked in a reliable way if our QA dependencies, mainly DrupalQA package, has support for that version.

See related story with PHP 8.0 when DrupalQA's minimum requirement for PHP 8.1 in #90.

### PR DESCRIPTION
**Pull request by mxr576**
also fix QA pipeline

Closes #93.

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -139,10 +139,16 @@
     },
     "scripts": {
         "post-install-cmd": [
-            "Pronovix\\SimpleSymlink\\ScriptHandler::createSymlinks"
+            "@symlink-for-local-dev-env"
```
Comment (mxr576) [inline composer.json:142]: As a replacement of `pronovix/simple-symlink`

**Comment by mxr576** [comment]
Current builds are failing due to issues related to [this old trick ](https://www.drupal.org/project/drupal/issues/3356283)that happened in older versions of Drupal.

https://github.com/Pronovix/swagger_ui_formatter/actions/runs/8597299839/job/23555697813

File: composer.json [hunk 2]
```diff
@@ -20,18 +20,17 @@
     ],
     "require": {
         "php": "~8.1.0",
```
Comment (dpagini) [inline composer.json:22]: I may be mistaken, but this won't "close #93" without addressing/increasing this line. Is this constraint even needed here? This is one of very few projects in Drupal that specifies such a tightly paired PHP constraint.

With Drupal 11 coming out _this year_, this needs to be elevated to PHP 8.3 before the end of the year. Maybe the constraint could at least be relaxed and start to allow for minor version bumps?

Comment (mxr576) [reply composer.json:22]: Thanks for catching this, must have been lost as part of rebasing commits... :facepalm: fixing it.

Adding Drupal 11 support requires a dedicated chapter. Small steps, release early :)

Comment (mxr576) [reply composer.json]: fixed.

Comment (dpagini) [reply composer.json]: Thanks @mxr576. To be fair, I'm not asking for Drupal 11 support... just to relax the PHP requirement on the project.

Comment (dpagini) [reply composer.json]: Again, there are not a lot of Drupal modules that do this... Couldn't you just say _any version of 8_ is **allowed** in your platform requirements... and then you can TEST and officially support 8.1 and 8.2 for example, but by saying _you can't use anything higher than 8.2_ you are really limiting anyone who wants to use this module.

**Review by lekob — APPROVED** [review]

**Pull request merged**